--- proxy_util.c	2017-12-21 16:53:02.203570945 +0800
+++ proxy_util.c	2017-12-21 17:11:24.505709197 +0800
@@ -2165,6 +2165,17 @@
     apr_status_t err = APR_SUCCESS;
     apr_status_t uerr = APR_SUCCESS;
     const char *uds_path;
+    const char *str_af = apr_table_get(r->subprocess_env, "proxy-address-family");
+    int af = APR_UNSPEC;
+    if(str_af) {
+        if(str_af[0] == '4') {
+            ap_log_error(APLOG_MARK, APLOG_DEBUG, 0, r->server, "proxy: ipv4 only");
+            af = APR_INET;
+        }else if(str_af[0] == '6'){
+            ap_log_error(APLOG_MARK, APLOG_DEBUG, 0, r->server, "proxy: ipv6 only");
+            af= APR_INET6;
+        }
+    }
 
     /*
      * Break up the URL to determine the host to connect to
@@ -2280,7 +2291,7 @@
                  * Otherwise we will look it up once for the worker.
                  */
                 err = apr_sockaddr_info_get(&(conn->addr),
-                                            conn->hostname, APR_UNSPEC,
+                                            conn->hostname, af,
                                             conn->port, 0,
                                             conn->pool);
             }
@@ -2305,7 +2316,7 @@
                  * inside dynamic config to force the lookup.
                  */
                 err = apr_sockaddr_info_get(&(worker->cp->addr),
-                                            conn->hostname, APR_UNSPEC,
+                                            conn->hostname, af,
                                             conn->port, 0,
                                             worker->cp->pool);
                 conn->addr = worker->cp->addr;
