--- mod_proxy_http.c	2017-10-25 02:30:20.130127922 +0800
+++ mod_proxy_http.c	2017-10-25 02:34:13.434117041 +0800
@@ -826,27 +826,30 @@
      */
 
     if (PROXYREQ_REVERSE == r->proxyreq) {
-        const char *buf;
+	const char *enable = apr_table_get(r->subprocess_env, "proxy-http-enable-x-forward");
+	if(enable && strcmp(enable,"1")==0) {
+		const char *buf;
 
-        /* Add X-Forwarded-For: so that the upstream has a chance to
-         * determine, where the original request came from.
-         */
-        apr_table_mergen(r->headers_in, "X-Forwarded-For",
-                         c->remote_ip);
+		/* Add X-Forwarded-For: so that the upstream has a chance to
+		 * determine, where the original request came from.
+		 */
+		apr_table_mergen(r->headers_in, "X-Forwarded-For",
+				 c->remote_ip);
 
-        /* Add X-Forwarded-Host: so that upstream knows what the
-         * original request hostname was.
-         */
-        if ((buf = apr_table_get(r->headers_in, "Host"))) {
-            apr_table_mergen(r->headers_in, "X-Forwarded-Host", buf);
-        }
+		/* Add X-Forwarded-Host: so that upstream knows what the
+		 * original request hostname was.
+		 */
+		if ((buf = apr_table_get(r->headers_in, "Host"))) {
+		    apr_table_mergen(r->headers_in, "X-Forwarded-Host", buf);
+		}
 
-        /* Add X-Forwarded-Server: so that upstream knows what the
-         * name of this proxy server is (if there are more than one)
-         * XXX: This duplicates Via: - do we strictly need it?
-         */
-        apr_table_mergen(r->headers_in, "X-Forwarded-Server",
-                         r->server->server_hostname);
+		/* Add X-Forwarded-Server: so that upstream knows what the
+		 * name of this proxy server is (if there are more than one)
+		 * XXX: This duplicates Via: - do we strictly need it?
+		 */
+		apr_table_mergen(r->headers_in, "X-Forwarded-Server",
+				 r->server->server_hostname);
+	}
     }
 
     proxy_run_fixups(r);
